http-client/src/test/groovy/io/micronaut/http/client/HttpClientTraceLoggingSpec.groovy

```diff
@@ -0,0 +1,66 @@
+package io.micronaut.http.client
+
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.AppenderBase
+import io.micronaut.context.ApplicationContext
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.runtime.server.EmbeddedServer
+import org.slf4j.LoggerFactory
+import spock.lang.Specification
+
+class HttpClientTraceLoggingSpec extends Specification {
+    def 'full response'() {
+        given:
+        String loggerName = 'io.micronaut.http.client.HttpClientTraceLoggingSpec.' + UUID.randomUUID()
+        MemoryAppender appender = new MemoryAppender()
+        Logger l = (Logger) LoggerFactory.getLogger(loggerName)
+        l.addAppender(appender)
+        l.setLevel(Level.ALL)
+        appender.start()
+
+        def configuration = new DefaultHttpClientConfiguration()
+        configuration.setLoggerName(loggerName)
+
+        def ctx = ApplicationContext.run()
+        def server = ctx.getBean(EmbeddedServer)
+        server.start()
+        def client = ctx.createBean(HttpClient, server.URI, configuration).toBlocking()
+
+        when:
+        client.exchange("/get-full")
+        then:
+        appender.events.any { it.contains('foo') }
+
+        cleanup:
+        client.close()
+        server.stop()
+        ctx.stop()
+    }
+
+
+    static class MemoryAppender extends AppenderBase<ILoggingEvent> {
```

Comment:
Instead of writing a MemoryAppender, we could use the ListAppender that comes with logback:

```diff
diff --git a/http-client/src/test/groovy/io/micronaut/http/client/HttpClientTraceLoggingSpec.groovy b/http-client/src/test/groovy/io/micronaut/http/client/HttpClientTraceLoggingSpec.groovy
index 28825d6348..1fdd470b0e 100644
--- a/http-client/src/test/groovy/io/micronaut/http/client/HttpClientTraceLoggingSpec.groovy
+++ b/http-client/src/test/groovy/io/micronaut/http/client/HttpClientTraceLoggingSpec.groovy
@@ -3,7 +3,7 @@ package io.micronaut.http.client
 import ch.qos.logback.classic.Level
 import ch.qos.logback.classic.Logger
 import ch.qos.logback.classic.spi.ILoggingEvent
-import ch.qos.logback.core.AppenderBase
+import ch.qos.logback.core.read.ListAppender
 import io.micronaut.context.ApplicationContext
 import io.micronaut.http.annotation.Controller
 import io.micronaut.http.annotation.Get
@@ -12,10 +12,11 @@ import org.slf4j.LoggerFactory
 import spock.lang.Specification

 class HttpClientTraceLoggingSpec extends Specification {
+
     def 'full response'() {
         given:
         String loggerName = 'io.micronaut.http.client.HttpClientTraceLoggingSpec.' + UUID.randomUUID()
-        MemoryAppender appender = new MemoryAppender()
+        def appender = new ListAppender<ILoggingEvent>()
         Logger l = (Logger) LoggerFactory.getLogger(loggerName)
         l.addAppender(appender)
         l.setLevel(Level.ALL)
@@ -32,30 +33,15 @@ class HttpClientTraceLoggingSpec extends Specification {
         when:
         client.exchange("/get-full")
         then:
-        appender.events.any { it.contains('foo') }
+        appender.list.formattedMessage.any { it.contains('foo') }

         cleanup:
+        appender.stop()
         client.close()
         server.stop()
         ctx.stop()
     }

-
-    static class MemoryAppender extends AppenderBase<ILoggingEvent> {
-        private final List<String> events = new ArrayList<>()
-
-        @Override
-        protected void append(ILoggingEvent e) {
-            synchronized (events) {
-                events.add(e.toString())
-            }
-        }
-
-        List<String> getEvents() {
-            return events
-        }
-    }
-
     @Controller
     static class Ctrl {
         @Get("/get-full")
```

Also, I think we need to stop the appender in cleanup